pkg/controller/daemon_controller.go

```diff
@@ -0,0 +1,432 @@
+				podsToDelete = append(podsToDelete, daemonPodNames[i])
+			}
+		} else if !shouldRun && isRunning {
+			// If daemon pod isn't supposed to run on node, but it is, delete all daemon pods on node.
+			for i := range daemonPodNames {
+				podsToDelete = append(podsToDelete, daemonPodNames[i])
+			}
+		}
+	}
+
+	// We need to set expectations before creating/deleting pods to avoid race conditions.
+	dcKey, err := controllerKeyFunc(dc)
+	if err != nil {
+		glog.Errorf("Couldn't get key for object %+v: %v", dc, err)
+	}
+	dm.expectations.SetExpectations(dcKey, len(nodesNeedingDaemons), len(podsToDelete))
+
+	glog.Infof("nodesNeedingDaemons: %+v", nodesNeedingDaemons)
+	for i := range nodesNeedingDaemons {
+		if err := dm.podControl.createReplicaOnNode(dc.Namespace, dc, nodesNeedingDaemons[i]); err != nil {
+			glog.V(2).Infof("Failed creation, decrementing expectations for controller %q/%q", dc.Namespace, dc.Name)
+			dm.expectations.CreationObserved(dcKey)
+			util.HandleError(err)
+		}
+	}
+
+	glog.Infof("podsToDelete: %+v", podsToDelete)
+	for i := range podsToDelete {
+		if err := dm.podControl.deletePod(dc.Namespace, podsToDelete[i]); err != nil {
+			glog.V(2).Infof("Failed deletion, decrementing expectations for controller %q/%q", dc.Namespace, dc.Name)
+			dm.expectations.DeletionObserved(dcKey)
+			util.HandleError(err)
+		}
+	}
+}
+
+func (dm *DaemonManager) updateDaemonStatus(dc *api.DaemonController) {
+	glog.Infof("Updating daemon status")
+	nodeToDaemonPods, err := dm.getNodesToDaemonPods(dc)
+	if err != nil {
+		glog.Errorf("Error getting node to daemon pod mapping for daemon %+v: %v", dc, err)
+	}
+
+	nodeList, err := dm.nodeStore.List()
+	if err != nil {
+		glog.Errorf("Couldn't get list of nodes when adding daemon %+v: %v", dc, err)
+	}
+
+	var desiredNumberScheduled, currentNumberScheduled, numberMisscheduled int
+	for i := range nodeList.Items {
+		nodeSelector := labels.Set(dc.Spec.Template.Spec.NodeSelector).AsSelector()
+		shouldRun := nodeSelector.Matches(labels.Set(nodeList.Items[i].Labels))
+		numDaemonPods := len(nodeToDaemonPods[nodeList.Items[i].Name])
+		if shouldRun {
+			desiredNumberScheduled++
+			if numDaemonPods == 1 {
+				currentNumberScheduled++
+			}
+		} else if numDaemonPods >= 1 {
+			numberMisscheduled++
+		}
+	}
+
+	glog.Infof("Counts: %d %d %d", desiredNumberScheduled, currentNumberScheduled, numberMisscheduled)
+	err = storeDaemonStatus(dm.kubeClient.DaemonControllers(dc.Namespace), dc, desiredNumberScheduled, currentNumberScheduled, numberMisscheduled)
+	if err != nil {
+		glog.Errorf("Error storing status for daemon %+v: %v", dc, err)
+	}
+}
+
+func (dm *DaemonManager) syncDaemonController(key string) error {
+	startTime := time.Now()
+	defer func() {
+		glog.V(4).Infof("Finished syncing daemon %q (%v)", key, time.Now().Sub(startTime))
+	}()
+	obj, exists, err := dm.dcStore.Store.GetByKey(key)
+	if !exists {
+		glog.Infof("Daemon Controller has been deleted %v", key)
```

Comment:
The default is 2, there is a convention doc written somewhere. Basically
we'd like to keep log files small by default.

On Thu, Jul 9, 2015 at 9:32 AM, Ananya Kumar <notifications@github.com>
wrote:

> In pkg/controller/daemon_controller.go
> <https://github.com/vmarmol/kubernetes/pull/2#discussion_r34274416>:
>
> > +
> > +	glog.Infof("Counts: %d %d %d", desiredNumberScheduled, currentNumberScheduled, numberMisscheduled)
> > +	err = storeDaemonStatus(dm.kubeClient.DaemonControllers(dc.Namespace), dc, desiredNumberScheduled, currentNumberScheduled, numberMisscheduled)
> > +	if err != nil {
> > +		glog.Errorf("Error storing status for daemon %+v: %v", dc, err)
> > +	}
> > +}
> > +
> > +func (dm *DaemonManager) syncDaemonController(key string) error {
> > +	startTime := time.Now()
> > +	defer func() {
> > +		glog.V(4).Infof("Finished syncing daemon %q (%v)", key, time.Now().Sub(startTime))
> > +	}()
> > +	obj, exists, err := dm.dcStore.Store.GetByKey(key)
> > +	if !exists {
> > +		glog.Infof("Daemon Controller has been deleted %v", key)
>
> Sure. What is our convention for leveled logging? What's the default level?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/vmarmol/kubernetes/pull/2/files#r34274416>.
>